server/core/region.go

```diff
@@ -670,6 +685,50 @@ func (r *RegionsInfo) RemoveRegion(region *RegionInfo) {
 	}
 }
 
+// shouldRemoveFromSubTree return true when the region leader changed, peer transferred,
+// new peer was created, learners changed, pendingPeers changed, and so on.
+func (r *RegionsInfo) shouldRemoveFromSubTree(region *RegionInfo, origin *RegionInfo) bool {
+	checkPeersChange := func(origin []*metapb.Peer, other []*metapb.Peer) bool {
+		checkPeersDiff := func(origin []*metapb.Peer, other []*metapb.Peer) bool {
```

Comment:
Do we have a better way (e.g. maybe sort them and compare them?) to do this check so that we can use something like `checkPeersDiff(origin, other)` rather than `checkPeersDiff(origin, other) || checkPeersDiff(other, origin)` which seems like do the same thing?